### DISCUSSION TITLE
Math block fence for GFM with Gitlab

### DISCUSSION THREAD

**Discussion by Yoshinobu-Ishizaki** [quarto-dev/quarto-cli#878] [Q&A] [answered]
I mainly use Gitlab instead of Github and want to generate Markdown file for Gitlab.

On Gitlab, math block must be fenced by ``` $` ``` and ``` `$ ``` for inline and ```` ```math ... ``` ```` for displayed ( See https://docs.gitlab.com/ee/user/markdown.html#math ).

This is a quick hack to generate `.md` file to fit well with Gitlab.

(1) Prepare with lua filter file (ex. `gitlab-quarto.lua`)
```
function Math(el)
    if el.mathtype == "InlineMath" then
        return {pandoc.RawInline("html","$`"), pandoc.RawInline("html",el.text), pandoc.RawInline("html","`$")}
    else
        return {pandoc.RawInline("html","\n\n```math"), pandoc.RawInline("html",el.text),pandoc.RawInline("html","\n```\n")}
    end
end
```

(2) Add YAML header 
```
  gfm:
    html-math-method: katex
    filters: [gitlab-quarto.lua]
```

I use katex because Gitlab uses it. 
Also, cross referencing math seems not working currently ( Codecogs does not accept `\tag` and generates error 😒  ) .

---

[Here](https://gitlab.com/yigrouptest/quarto_test/-/blob/main/gfm_math_test.md) is a sample.

It would be nice if this feature is included officially 😄 .

---

Quarto 0.9.402 , RStudiio 2022-02, R-4.1.2

**Comment by jjallaire** [comment, marked as answer]
I think we are overzealous w/ our use of `\tag`. This commit (https://github.com/quarto-dev/quarto-cli/commit/511685e11771a983c2047a1c96cdd97219490692) goes back to `\qquad` for any scenario (including docx) which isn't mathjax or katex.
